### ISSUE TITLE
[pg-cursor] When do you need rowCount other than 1

### ISSUE THREAD

**Issue by the-spyke** [brianc/node-postgres#2868]
Hello, I want to understand what is the difference in performance or other implications of various rowCount values for `pg-cursor`.

If it is streaming, then rows arrive one by one and just put inside an array in the pg-cursor. I can do this myself, especially when a cursor is wrapped into an async iterable.

It would be nice to have a paragraph in the docs about this.

**Comment by brianc** [comment]
Yeah good call should add something to docs.  The basic gist is the cursor rowCount size directly impacts performance.  Each "page" you fetch from the cursor is a round-trip of network packets to and from your app -> postgres -> back again.  So a rowCount of `1` would be the absolute least efficent value you could use there.  Typically I'd look for a more reasonable _upper_ bound and that's more to do w/ the row size (800 columns in the result set you're gonna wanna bring back fewer rows per page since each page is held in memory while its consumed in your app).  TBH `100` to `1000` is probably something you can't really go wrong with....but wouldn't really ever use `1` except for like testing that the cursor works or something.

**Comment by the-spyke** [comment]
So, even if the parameter of the `client.query()` is called `queryStream`, the cursor in not like a Node stream with high/low watermarks which pre-buffers some data for the reader. A cursor is just a `pull` mechanism. And you need to find an optimal *page* size for your particular query based on latency and memory. Thank you!

**Comment by brianc** [comment]
`queryStream` has some built in [sane defaults](https://github.com/brianc/node-postgres/blob/master/packages/pg-query-stream/src/index.ts#L25) but yea...postgres only lets you read rows two ways: 1) pull. 2) firehose.
